### DISCUSSION TITLE
Can Stylus be applied to the Stylus editor (CodeMirror)?  It used to work.

### DISCUSSION THREAD

**Discussion by sasgrw** [openstyles/stylus#1580] [Q&A]
I have the following line in my Stylus style:

`
.CodeMirror-line {font-size:1.4em !important}
`

This used to increase the fontsize when I was using the editor in Stylus but it no longer works.  I think the last time I noticed it working was maybe a month ago, although my timing could be off. I know it used to work.  Has anything changed in Stylus itself so that it doesn't apply styles to the Stylus editor?

If I view a page where Stylus **is** be applied, I can see my Stylus styles being included at the end of the page via a `<style>` element.  This `<style>` element does **not** exist when I view the code inspector for the Stylus editor so my Stylus styles are not being included.

Was this an intentional change?

**Comment by siffegh** [comment]
.CodeMirror-line {font-size:1.4em !important} works for me, as does .CodeMirror {font-size: 15px !important;}.

**Reply by tophf** [reply]
Confirming. Still works. There's no change.

**Comment by sasgrw** [comment]
I just noticed that the Stylus icon in my plugin list is turned off when I'm in the Stylus editor. If I try to turn it on, it says "No styles installed for this site."

![image](https://user-images.githubusercontent.com/9466851/230034794-fbdc983e-170e-4a75-b140-fbe276e32b92.png)

That would explain why my style isn't being applied but why does Stylus turn itself off when I'm editing the style?

**Reply by tophf** [reply]
It depends on what exactly you are doing. Maybe you can make a screen recording.

**Reply by sasgrw** [reply]
Yes, I can make a screen recording but I'm not doing anything complicated. I have the plugin installed with a custom style sheet. It works great in all websites I visit. It used to work in the Stylus editor itself but as of recently, the plugin turns itself off with the above message when I try to edit my style. The plugin icon goes from a bright cyan to a dimmed cyan (like a disabled button).

**Reply by sasgrw** [reply]
For example, on this github page, the plugin icon looks like this:

![image](https://user-images.githubusercontent.com/9466851/230046782-d2d9d699-38d5-4535-8ff2-999d7fe9bb97.png)

But in the Stylus editor, it looks like this:

![image](https://user-images.githubusercontent.com/9466851/230046850-ced019c4-775d-4962-9adf-a3dc78c98039.png)

**Reply by sasgrw** [reply]
I looked through all the Options and didn't see anything that might affect it.

![image](https://user-images.githubusercontent.com/9466851/230047878-7c0f3093-8b60-403e-bd0c-b296ebce262a.png)

**Reply by tophf** [reply]
I can't guess what happens based on your description. I'd need to see the exact sequence and the code of the style.

**Reply by tophf** [reply]
Hmm, wait, I have an idea: you may have excluded the editor accidentally:

![image](https://user-images.githubusercontent.com/1310400/230048737-5f11f129-cb0e-4cdd-9a8a-00a23927e487.png)

**Reply by sasgrw** [reply]
Video attached.


https://user-images.githubusercontent.com/9466851/230049561-47fe9005-1934-4b9f-838e-33f73ddaee0c.mp4

**Reply by sasgrw** [reply]
There is no domain to exclude. I'm in the Stylus editor and there isn't a 3-vertical-dot menu in the Stylus dialog as seen in the video.

**Comment by tophf** [comment]
Judging by the video you're using a global style i.e. without site qualifiers. Previously such styles were mistakenly applied to the Stylus' own UI, consequently making it unusable. We fixed this bug, so global styles currently apply only to the web sites.

To style Stylus editor or any other page of Stylus extension you need to do it in a section that explicitly targets the extension e.g. you can use URL prefix `moz-extension://` and/or `chrome-extension://`.

**Reply by sasgrw** [reply]
That's too bad. I took advantage of that feature, having it applied to Stylus' UI, but it sounds like there's a workaround. I'll have to look up how to specify the prefix in the CSS.

When was this "bug" fixed? That would at least give me a timeframe for when I last saw this working.

**Reply by tophf** [reply]
If you want to use the extension the wrong way you'll have to hack it on your own.

The correct way to target a page is to create a new section and specify its target.

If the section is global like yours there'll be a `+` button:

![image](https://user-images.githubusercontent.com/1310400/230136954-a0e87176-9bd0-4e33-9d60-8a79187be77e.png)

click it and then set the target:

![image](https://user-images.githubusercontent.com/1310400/230137549-751502e7-2b44-4d36-82a6-dd6c571572cd.png)

**Reply by sasgrw** [reply]
>If you want to use the extension the wrong way you'll have to hack it on your own.

C'mon now. I'm a developer too and I've had tons of times when users used my code in a way I didn't expect.

I looked at the change log, https://github.com/openstyles/stylus/wiki/Log, but couldn't find when this change happened. Would you mind sharing that?

**Reply by tophf** [reply]
I don't remember.
